java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JdbcUtils.java

```diff
@@ -43,6 +43,18 @@ public static Connection getConnection(String jdbcUrl) throws SQLException {
         // https://jdbc.postgresql.org/documentation/use/
         // https://dev.mysql.com/doc/connectors/en/connector-j-connp-props-networking.html#cj-conn-prop_tcpKeepAlive
         props.setProperty("tcpKeepAlive", "true");
+
+        // default timeout in seconds
+        final int CONNECTION_TIMEOUT = 30;
+        final int SOCKET_TIMEOUT = 300;
+        boolean isPg = jdbcUrl.startsWith("jdbc:postgresql");
+
+        // postgres use seconds and mysql use milliseconds
+        int connectTimeout = isPg ? CONNECTION_TIMEOUT : CONNECTION_TIMEOUT * 1000;
+        int socketTimeout = isPg ? SOCKET_TIMEOUT : SOCKET_TIMEOUT * 1000;
+        props.setProperty("connectTimeout", String.valueOf(connectTimeout));
```

Comment:
Instead of passing the value via props, I think we can create the connection with only `jdbcUrl` anyway and then get and reset the timeout by the methods provided in the `Connection`

https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/sql/Connection.html

It provides get/setNetworkTimeout method in the interface.

Reply:
> [setNetworkTimeout](https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/sql/Connection.html#setNetworkTimeout-java.util.concurrent.Executor-int-)([Executor](https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/util/concurrent/Executor.html) executor, int milliseconds)

I inspected the interface. But the first arg is confusing. For PG we can pass null for the `executor`, but for MySQL it will throw an exception if first arg is null.